hdf5-rs/src/hl/container.rs

```diff
@@ -261,18 +277,41 @@ impl<'a> Writer<'a> {
         } else {
             let fspace = self.obj.space()?;
             let slice_shape = fspace.select_slice(slice)?;
-            let mspace = Dataspace::try_new(&slice_shape, false)?;
 
             let view = arr.into();
+            let data_shape = view.shape();
+
+            // Restore dimensions that are SliceOrIndex::Index in the slice.
+            let mut data_shape_hydrated = Vec::new();
+            let mut pos = 0;
+            for s in slice_s {
+                match s {
+                    SliceOrIndex::Index(_) => {
+                        data_shape_hydrated.push(1);
+                    }
+                    _ => {
+                        data_shape_hydrated.push(data_shape[pos]);
+                        pos += 1;
+                    }
+                }
+            }
+
+            let mspace = Dataspace::try_new(&slice_shape, false)?;
 
             // FIXME - we can handle non-standard input arrays by creating a memory space
             // that reflects the same slicing/ordering that this ArrayView represents.
```

Comment:
Wondering - how hard would it be to support non standard layouts? As long as it's convertible to a hyperslab spec, it should be fine, no?

I'm thinking maybe all this "hyperslab" <--> "ndarray indexing" business should be moved out into a separate API to simplify all this logic here (and tested separately, without even touching HDF5 stuff). Not saying it should be done in this PR, just generally/later. Thoughts?

Reply:
yes, definitely - should be doable, and should be abstracted out of this code. Filed as #21.